### PR TITLE
updated test_param_scheduler replacing multiple calls with pytest.parametrize

### PR DESCRIPTION
**Pull request by Ishan-Kumar2**
Addresses #2522

Description:
Updated handlers/test_param_scheduler replacing multiple calls with pytest.parametrize

Haven't converted 

https://github.com/pytorch/ignite/blob/69ea3c8fef56191335771665c6acb9937f7a8668/tests/ignite/handlers/test_param_scheduler.py#L812 
and https://github.com/pytorch/ignite/blob/69ea3c8fef56191335771665c6acb9937f7a8668/tests/ignite/handlers/test_param_scheduler.py#L1228

because it has conditional calls. We could have different fixtures for all the cases but I believe that would make the code less readable, let me know what you think would be best for these tests.
Check list:

- [X] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/handlers/test_param_scheduler.py [hunk 1]
```diff
@@ -637,66 +632,69 @@ def test_lr_scheduler_asserts():
...
 
 
-def test_lr_scheduler():
-    def _test(torch_lr_scheduler_cls, **kwargs):
+@pytest.mark.parametrize(
+    "torch_lr_scheduler_cls, kwargs",
+    [
+        (StepLR, ({"step_size": 5, "gamma": 0.5})),
+        (ExponentialLR, ({"gamma": 0.78})),
+        (MultiplicativeLR, ({"lr_lambda": lambda epoch: 0.95})),
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_param_scheduler.py:640]: This wont work as older versions wont be able to import the class

**Comment by vfdev-5** [comment]
@Ishan-Kumar2 thanks for the PR, I left a comment to address.

> We could have different fixtures for all the cases but I believe that would make the code less readable, let me know what you think would be best for these tests.

I agree, let's keep them as they are now...

File: tests/ignite/handlers/test_param_scheduler.py (hunk 1 above)
Comment (Ishan-Kumar2) [reply tests/ignite/handlers/test_param_scheduler.py]: I have added a fix, please have a look thanks!

**Review by vfdev-5 — APPROVED** [review]
@Ishan-Kumar2 LGTM, thanks for the PR !

**Pull request merged**